concreate/tools.py

```diff
@@ -94,27 +95,87 @@ def load_descriptor(descriptor_path, schema_type):
     logger.debug("Loading %s descriptor from path '%s'."
                  % (schema_type,
                     descriptor_path))
-    schema_name = '%s_schema.yaml' % schema_type
-    schema_path = os.path.join(os.path.dirname(__file__),
-                               'schema',
-                               schema_name)
-    if not os.path.exists(schema_path):
-        raise ConcreateError('Cannot locate schema for %s.' % schema_type)
 
+    if not os.path.exists(descriptor_path):
+        raise ConcreateError('Cannot find provided descriptor file')
+
+    common_schema_path = os.path.join(os.path.dirname(__file__),
+                               'schema',
+                               'common.yaml')
     schema = {}
-    with open(schema_path, 'r') as fh:
+
+    # Read main schema definition
+    with open(common_schema_path, 'r') as fh:
         schema = yaml.safe_load(fh)
 
-    if not os.path.exists(descriptor_path):
-        raise ConcreateError('Cannot find provided descriptor file')
+    specific_schema_path = os.path.join(os.path.dirname(__file__),
+                                        'schema',
+                                        '%s.yaml' % schema_type)
+
+    # Read schema definition for specific type
+    with open(specific_schema_path, 'r') as fh:
+        specific_schema = yaml.safe_load(fh)
+        schema = merge_dictionaries(schema, specific_schema)
 
     descriptor = {}
+
     with open(descriptor_path, 'r') as fh:
         descriptor = yaml.safe_load(fh)
 
-    core = Core(source_data=descriptor, schema_data=schema)
+    core = Core(source_data=descriptor, schema_data=schema, allow_assertions=True)
     try:
         return core.validate(raise_exception=True)
     except Exception as ex:
         raise ConcreateError("Cannot validate schema: %s" % (descriptor_path),
                              ex)
+
+
+def merge_dictionaries(dict1, dict2):
+    """ Merges two dictionaries handling embedded lists and
+    dictionaries.
+    In a case of simple type, values from dict2 are preserved.
+
+    Args:
+      dict1, dict2 dictionaries to merge
+
+    Return merged dictionaries
+    """
+    for k2, v2 in dict2.items():
+        if k2 not in dict1:
+            dict1[k2] = v2
+        else:
+            if isinstance(v2, list):
+                dict1[k2] = merge_lists(dict1[k2], v2)
+            elif isinstance(v2, dict):
+                dict1[k2] = merge_dictionaries(dict1[k2], v2)
+            else:
+                # if the type is int or strings we override the value from
+                # dict2
+                dict1[k2] = v2
```

Comment:
@dbecvarik Here is the change.

Reply:
Isn't this regression for https://github.com/goldmann/concreate/blob/a1a2bea60a7b82917361a2b896267eeceec7c918/concreate/descriptor.py#L78
Especially for name key? So we will override image.yaml `name` with a module.yaml `name`?

Reply:
Good catch!

Reply:
Fixed!